src/utilities/paramValidation.ts

```diff
@@ -1,18 +1,31 @@
-import { Param, RouteMatchRule } from '@/types'
+import { ResolvedRoute, RouteMatchRule } from '@/types'
 import { getParamValue, getParamValuesFromUrl } from '@/utilities'
 
 export const routeParamsAreValid: RouteMatchRule = (route, url) => {
-  const params = Object.entries<Param[]>(route.params)
+  try {
+    getRouteParams(route, url)
+  } catch {
+    return false
+  }
+
+  return true
+}
 
-  for (const [key, paramsTuple] of params) {
+export const getRouteParams = (route: ResolvedRoute, url: string): Record<string, unknown> => {
+  const params: Record<string, unknown> = {}
+
+  for (const [key, paramsTuple] of Object.entries(route.params)) {
     const stringValues = getParamValuesFromUrl(url, route.path, key)
+    const values = paramsTuple.map((param, index) => getParamValue(stringValues[index], param))
 
-    try {
-      paramsTuple.forEach((param, index) => getParamValue(stringValues[index], param))
-    } catch {
-      return false
+    if (paramsTuple.length === 1) {
+      const [value] = values
+      params[key] = value
+      continue
     }
```

Comment:
Note that if only a single param is expected its "unwrapped"

Reply:
`nit`, I always try to avoid "continue". How would you feel about just using `else` instead?

```ts
if (paramsTuple.length === 1) {
  const [value] = values
  params[key] = value
} else {
  params[key] = values
}
```

Reply:
Going to prefer continue over else to follow the "return early" pattern.